homeassistant/components/trend/binary_sensor.py

```diff
@@ -44,30 +51,66 @@
     CONF_INVERT,
     CONF_MAX_SAMPLES,
     CONF_MIN_GRADIENT,
+    CONF_MIN_SAMPLES,
     CONF_SAMPLE_DURATION,
     DOMAIN,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
-SENSOR_SCHEMA = vol.Schema(
-    {
-        vol.Required(CONF_ENTITY_ID): cv.entity_id,
-        vol.Optional(CONF_ATTRIBUTE): cv.string,
-        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
-        vol.Optional(CONF_FRIENDLY_NAME): cv.string,
-        vol.Optional(CONF_INVERT, default=False): cv.boolean,
-        vol.Optional(CONF_MAX_SAMPLES, default=2): cv.positive_int,
-        vol.Optional(CONF_MIN_GRADIENT, default=0.0): vol.Coerce(float),
-        vol.Optional(CONF_SAMPLE_DURATION, default=0): cv.positive_int,
-    }
+
+def _check_sample_options(config: ConfigType) -> ConfigType:
+    """Check min/max sample options."""
+    if config[CONF_MAX_SAMPLES]:
+        if config[CONF_MAX_SAMPLES] < config[CONF_MIN_SAMPLES]:
+            raise vol.Invalid(
+                f"{CONF_MAX_SAMPLES} must not be smaller than {CONF_MIN_SAMPLES}"
+            )
+    elif not config[CONF_SAMPLE_DURATION]:
+        raise vol.Invalid(
+            f"{CONF_MAX_SAMPLES} & {CONF_SAMPLE_DURATION} cannot both be zero"
+        )
+    else:
+        config[CONF_MAX_SAMPLES] = None
+    return config
+
+
+SENSOR_SCHEMA = vol.All(
+    vol.Schema(
+        {
+            vol.Required(CONF_ENTITY_ID): cv.entity_id,
+            vol.Optional(CONF_UNIQUE_ID): cv.string,
+            vol.Optional(CONF_ATTRIBUTE): cv.string,
+            vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+            vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+            vol.Optional(CONF_INVERT, default=False): cv.boolean,
+            vol.Optional(CONF_MAX_SAMPLES, default=2): vol.All(
+                vol.Coerce(int),
+                vol.Any(0, vol.Range(min=2), msg="must be 0 or 2 or larger"),
+            ),
+            vol.Optional(CONF_MIN_GRADIENT, default=0.0): vol.Coerce(float),
+            vol.Optional(CONF_MIN_SAMPLES, default=2): vol.All(
+                vol.Coerce(int), vol.Range(min=2)
+            ),
+            vol.Optional(CONF_SAMPLE_DURATION, default=0): cv.positive_time_period,
+        }
```

Comment:
Based on [ADR-0007](https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md) you are not allowed to make changes to YAML configuration in platform sections until the integration has been refactored and the configuration moved under the integration domain key.

Reply:
From reading ADR-0010, I think this integration would be fine to keep a yaml config, and so should be fine as an exception to 0007?